### PR TITLE
[doc] Update secure hw design guidelines

### PR DESCRIPTION
**Pull request by msfschaffner**
I just realized that we never added these two points that were originally suggested by @cdgori after going through all the D2S revies.

The PR also adds more links pointing to the secure hw design guidelines so that they are more readily discoverable.

### CODE REVIEW

File: doc/security/implementation_guidelines/hardware/index.md [hunk 1]
```diff
@@ -214,6 +214,16 @@ attacks, see [[12](#ref-12)] and [[13](#ref-13)])
...
     recommend a default-safe posture, but each decision about handling detected
     faults should be carefully considered.
+16. For request-acknowledge interfaces, monitor the acknowledge line for
+    spurious pulses at all times (not only when pending request) and use this
+    as a glitch/fault detector to escalate locally and/or generate alerts.
+17. When arbitrating between two or more transaction sources with different
+    privilege/access levels, consider how to protect a request from one source
+    being glitched/forged to masquerade as being sourced from another
+    higher-privilege source (for example, to return side-loaded
+    hardware-visible-only data via a software read path).  At a minimum,
```
Comment (marnovandermaas) [inline doc/security/implementation_guidelines/hardware/index.md:224]: Nit. Two spaces in front of "At"

File: doc/ug/design.md [hunk 2]
```diff
@@ -30,6 +30,11 @@ TODO: briefly discuss key architectural decisions, and how we came to the conclu
 *   Bus strategy
 *   Reset strategy
 
+## Secure Hardware Design Guidelines
+
+Silicon designs for security devices require special guidelines to protect the designs against myriad attacks.
+To that end, the team established [Secure Hardware Design Guidelines]({{< relref "doc/security/implementation_guidelines/hardware" >}}) which are followed when developing OpenTitan security IP.
+
```
Comment (marnovandermaas) [inline doc/ug/design.md:37]: Is it desirable to have an exact copy of the section you added in `doc/security/_index.md` here?

Comment (msfschaffner) [reply doc/ug/design.md:37]: The idea was to make this page better discoverable and this page here is the overview page that has several security related pages underneath (in the folder structure).

When you go to this page today (https://docs.opentitan.org/doc/security/) it is a bit odd to see a link to "Implementation Guidelines" on the left side, but no associated section on the overview page itself.

File: doc/security/implementation_guidelines/hardware/index.md (hunk 1 above)
Comment (msfschaffner) [reply doc/security/implementation_guidelines/hardware/index.md]: Fixed - thanks!

File: doc/ug/design.md (hunk 2 above)
Comment (marnovandermaas) [reply doc/ug/design.md:37]: Maybe add it as a link to the `Language and Tool Selection` section instead of making it it's own section?

Comment (marnovandermaas) [reply doc/ug/design.md:37]: Also, maybe we can add a link on this page as well? https://docs.opentitan.org/doc/ug/hw_design/

Comment (msfschaffner) [reply doc/ug/design.md]: OK, amended accordingly - PTAL.

**Review by marnovandermaas — APPROVED** [review]
LGTM, thanks!

**Review by cdgori — COMMENTED** [review]
post-merge but LGTM - thanks for adding these 2 points and pushing up the visibility of the doc.

**Pull request merged**